### PR TITLE
Setup join/cap proc

### PR DESCRIPTION
**Pull request by bdero**
For adding the rest of the caps and joins.
Part of https://github.com/flutter/flutter/issues/99089.

### CODE REVIEW

File: entity/contents.h [hunk 1]
```diff
@@ -119,16 +121,26 @@ class SolidStrokeContents final : public Contents {
...
   };
 
   enum class Join {
     kMiter,
     kRound,
     kBevel,
-    kLast,
   };
 
+  typedef std::function<void(
```
Comment (chinmaygarde) [inline entity/contents.h:132]: Please prefer `using` instead of `typedef` unless the TU is meant to be C.

File: entity/contents.cc [hunk 2]
```diff
@@ -282,7 +282,10 @@ const IRect& TextureContents::GetSourceRect() const {
  ******* SolidStrokeContents
  ******************************************************************************/
 
-SolidStrokeContents::SolidStrokeContents() = default;
+SolidStrokeContents::SolidStrokeContents() {
+  SetStrokeCap(Cap::kButt);
+  SetStrokeJoin(Join::kBevel);
```
Comment (dnfield) [inline entity/contents.cc:287]: For Skia and Flutter, the default is `kMiter`.

Comment (dnfield) [reply entity/contents.cc:287]: I see miter isn't implemented yet, we should have a TODO here at least to default this to kMiter.

Comment (bdero) [reply entity/contents.cc:287]: Yeah, but I need to set it to kBevel by default until kMiter is implemented (coming in a few mins).

**Review by dnfield — COMMENTED** [review]
tests?

**Review by chinmaygarde — APPROVED** [review]

File: entity/contents.h (hunk 1 above)
Comment (bdero) [reply entity/contents.h]: Done.

File: entity/contents.cc (hunk 2 above)
Comment (bdero) [reply entity/contents.cc:288]: Done.

**Comment by bdero** [comment]
I added a test for the default setting and added a TODO for checking [Get|Set]Stroke[Join|Cap] properly once there are multiple to choose from.

**Pull request merged**